File: cpp/ecbenchmark/src/cec10/F01.cpp
#include "ecbenchmark/cec10/F01.h"
#include "ecbenchmark/cec10/CecMath.h"
#include "ecbenchmark/operator/Shifted.h"
#include "ecbenchmark/function/Elliptic.h"
#include "ecbenchmark/util/CommonMath.h"
namespace ecb {
    namespace cec10 {

        F01::F01(int dimensions)
        : CecFunction("F01", "Shifted Elliptic Function", dimensions,
        -100, 100, true, 0, new CecRandom(1l)) {
            std::vector<scalar> shift = CecMath::ShiftVector(dimensions, minimumDomain(), maximumDomain(), randomizer());
            elliptic = new Shifted(shift, 0, new Elliptic());
        }

        F01::~F01() {
            delete elliptic;
        }

        scalar F01::f(const std::vector<scalar>& x) {
            return elliptic->f(x);
        }
    }
}

File: java/ecbenchmark/src/ecbenchmark/cec10/F01.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.function.Elliptic;
import ecbenchmark.Function;
import ecbenchmark.operator.Shifted;

/**
 *
 * @author jcrada
 */
public class F01 extends CecFunction {

    private Function elliptic;

    public F01(int dimensions) {
        super("F01", "Shifted Elliptic Function", dimensions,
                -100, 100, true, new CecRandom(1l), 0);
        double[] shift = CecMath.ShiftVector(dimensions, getMinimum(), getMaximum(), getRandomizer());
        elliptic = new Shifted(shift, 0, new Elliptic());
    }

    public double f(final double[] x) {
        return elliptic.f(x);
    }
}